vault/cors.go

```diff
@@ -0,0 +1,137 @@
+package vault
+
+import (
+	"errors"
+	"net/http"
+	"regexp"
+	"strings"
+)
+
+var errCORSNotConfigured = errors.New("CORS is not configured")
+
+type CORSConfig struct {
+	enabled          bool
+	allowedOrigins   *[]string
```

Comment:
That's a coding-style thing. I'd rather pass pointers around, because I can check for nil.

I'd say I'm an intermediate novice when it comes to Go, so if idiomatically it shouldn't be let me know.